### ISSUE TITLE
`Unhandled Promise rejection` errors are thrown when adding basemaps to existing BasemapGalleryWidget

### ISSUE THREAD

**Issue by bpo19** [Esri/angular-cli-esri-map#109]
`Unhandled Promise rejection` errors are thrown in a simple Angular 12 application using esri-loader 3.1.0 and ArcGIS JS API 4.19 
when adding basemaps to existing BasemapGalleryWidget. 

A simple application that demonstrates the issue can be found here: https://github.com/bpo19/arcgis-js-sample-basemaps.
Errors are thrown when adding additional basemaps to the widget that was already created: `basemapGallery.source.basemaps.add(basemap);`
For large number of basemaps the console is flooded with errors and any other relevant information can be easily missed.

The errors started to appear in ArcGIS JS API 4.18 and are present in the next version (4.20) as well.

In order to reproduce the bug we followed the Esri sample from https://github.com/Esri/jsapi-resources/tree/master/esm-samples/jsapi-angular-cli
where we replaced the @arcgis/core library with esri-loader, as well as the AMD modules sample from https://github.com/Esri/angular-cli-esri-map.

**Comment by andygup** [comment]
Hi @bpo19 I was able to repro the issue with esri-loader but not with @arcgis/core. We're taking a closer look.

**Comment by andygup** [comment]
@bpo19 We've identified a fix and will try to get it into the up coming 4.20 AMD CDN release. That release is expected out in a couple weeks.

**Comment by andygup** [comment]
@bpo19 you can verify that the fix is in the up coming 4.20 AMD CDN release by adding this script tag to the Angular project's index.html:

```html
<script src="https://js.arcgis.com/next/" data-esri-loader="loaded"></script>
```
References:
* https://github.com/Esri/esri-loader#using-your-own-script-tag 
* https://github.com/Esri/feedback-js-api-next

**Comment by bpo19** [comment]
Thanks @andygup, in 4.20 I don't see the errors.
